src/js/stores/NewsFeedStore.jsx

```diff
@@ -78,22 +84,21 @@ class NewsFeedStore extends EventEmitter {
       failure: function(error){
         dispatcher.dispatch({type: "NEWSFEED_ERROR", error: error});
       }
-
     });
-
   }
 
 
   handleActions(action) {
     switch(action.type) {
       case "NEWSFEED_POST": {
-        // TODO: Readd this back in
-        this.emit("updated");
+        // TODO: ensure that client sees their post on the front end
+        this.emit("post");
         break;
       }
       case "NEWSFEED_GET": {
-        this.posts = action.posts;
-        // TODO: Append posts, don't simply delete old data
+        Array.prototype.push.apply(this.contentFeed.items, action.response.data.posts[0]);
```

Comment:
I'm not a huge fan of how I did this either and am very open to other ways of paginating. I haven't really implemented it before. The GET will almost always get more than 1 post but push.apply pretty much appends one array to the other. (and unlike concat, it doesn't create a new array)
https://davidwalsh.name/merge-arrays-javascript